lib/matplotlib/figure.py

```diff
@@ -2308,20 +2326,48 @@ def __init__(self,
 
         self.subplotpars = subplotpars
 
-        # constrained_layout:
-        self._constrained = False
-
-        self.set_tight_layout(tight_layout)
-
         self._axstack = _AxesStack()  # track all figure axes and current axes
         self.clf()
         self._cachedRenderer = None
 
-        self.set_constrained_layout(constrained_layout)
-
         # list of child gridspecs for this figure
         self._gridspecs = []
 
+    def set_layout_engine(self, layout=None, **kwargs):
```

Comment:
This now does:

```python
if self._check_layout_engines_compat(self._layout_engine,
                                             new_layout_engine):
            self._layout_engine = new_layout_engine
        else:
            _api.warn_external('Colorbar layout of new layout engine not '
                               'compatible with old engine, and a colorbar '
                               'has been created.  Engine not changed.')
```
which allows the engine to change if it can.

The most _useful_ case is if the user wants to try a few different params for the layout engine.   An advantage of a class is not only that we can pass other classes, but that each class can have a different set of inits/setters and the user may want to toggle these (pads, layout options etc).